tests/notificationprofile/test_views.py

```diff
@@ -40,113 +38,236 @@ def setUp(self):
-        self.email_destination = self.create_email_destination()
-
-        email_destination_pk = self.email_destination.pk
-        email_destination_path = reverse(
-            "v2:notification-profile:destinationconfig-detail", args=[email_destination_pk]
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
+        response = self.user1_rest_client.delete(
+            path=f"/api/v2/notificationprofiles/destinations/{self.non_synced_email_destination.pk}/"
         )
-
-        self.assertEqual(self.user1.destinations.get(pk=email_destination_pk), self.email_destination)
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_200_OK)
-        response = self.user1_rest_client.delete(email_destination_path)
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
-
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_404_NOT_FOUND)
+        self.assertFalse(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
 
     def test_cannot_delete_synced_email_destination(self):
-        self.email_destination = self.user1.destinations.filter(media_id="email").filter(settings__synced=True).first()
-
-        email_destination_path = reverse(
-            "v2:notification-profile:destinationconfig-detail", args=[self.email_destination.pk]
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=True).exists())
+        response = self.user1_rest_client.delete(
+            path=f"/api/v2/notificationprofiles/destinations/{self.synced_email_destination.pk}/"
         )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=True).exists())
 
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_200_OK)
-        response = self.user1_rest_client.delete(email_destination_path)
+    def test_cannot_delete_connected_email_destination(self):
+        self.notification_profile1.destinations.add(self.non_synced_email_destination)
+
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
+        response = self.user1_rest_client.delete(
+            path=f"/api/v2/notificationprofiles/destinations/{self.non_synced_email_destination.pk}/"
+        )
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
 
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_200_OK)
+    def test_can_get_incidents_of_notification_profile(self):
```

Comment:
Probably add an "all" here
```suggestion
    def test_can_get_all_incidents_of_notification_profile(self):
```